submissions/lalalilulu/tiny-world-pre-oop/index.js

```diff
@@ -0,0 +1,77 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/lalalilulu/a-tiny-JS-world
+   Web app: https://lalalilulu.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+
+    constructor(species, name, gender, saying, friends = []) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.friends = Array.from(friends).join(", ");
+    }
+
+    introduce() {
+        return`${this.saying}, I am a ${this.species}, my name is ${this.name}, my gender is ${this.gender}. My friends: ${this.friends}`;
+    }
+}
+
+class Human extends Inhabitant{
+
+    constructor(name, gender, saying, friends, legs = 2, hands = 2) {
+        super('human', name, gender, saying, friends);
+        this.legs = legs;
+        this.hands = hands;
+    }
+
+    introduce() {
+        return  super.introduce() + ` and I have ${this.legs} legs and ${this.hands} hands`;
+    }
+}
+
+class Animal extends Inhabitant{
+
+    constructor(species, name, gender, saying, friends, legs = 4) {
+        super(species, name, gender, saying, friends);
+        this.legs = legs;
+    }
+
+    introduce() {
+        return  super.introduce() + ` and I have ${this.legs} legs`;
+    }
+}
+
+const dog = new Animal('dog', 'Toby', 'male', 'woof-woof!', ["Kate", "Greg"]);
+const cat = new Animal('cat', 'Mary', 'female', 'meow-meow!', [dog.name]);
+const man = new Human( 'Fred', 'male', 'Hi!', [dog.name, cat.name]);
+const woman = new Human('Katy', 'female', 'Hello!', [dog.name, cat.name, man.name]);
```

Comment:
Not required to fix within the pre-OOP. But will be required to fix later in post-OOP task.

Here is the problem:
Taxonomy doesn't look really consistent.
```
- Inhabitant
  |- Human
  \- Animal
```
Human represents a specific species, while Animal is not. Besides, humans are animals as well.

There are multiple ways to implement a hierarchy of classes representing domain taxonomy, but for consistency each level of hierarchy should reflect the same level of abstraction.

Reply:
Thanks for the detailed explanation - I have changed classes hierarchy     
        Inhabitant
              | 
          Animal 
          /\     \
Human Dog  Cat